rapids-cmake/export/package.cmake

```diff
@@ -72,7 +77,18 @@ function(rapids_export_package type name export_set)
     endif()
   endif()
 
-  if(_RAPIDS_VERSION)
+  if(_RAPIDS_COMPONENTS AND _RAPIDS_VERSION)
+    set(version ${_RAPIDS_VERSION})
+    set(components ${_RAPIDS_COMPONENTS})
+    configure_file("${CMAKE_CURRENT_FUNCTION_LIST_DIR}/template/${type}_package_components_versioned.cmake.in"
+                   "${CMAKE_BINARY_DIR}/rapids-cmake/${export_set}/${type}/package_${name}.cmake"
+                   @ONLY)
+  elseif(_RAPIDS_COMPONENTS)
+    set(components ${_RAPIDS_COMPONENTS})
+    configure_file("${CMAKE_CURRENT_FUNCTION_LIST_DIR}/template/${type}_package_components.cmake.in"
+                   "${CMAKE_BINARY_DIR}/rapids-cmake/${export_set}/${type}/package_${name}.cmake"
+                   @ONLY)
+  elseif(_RAPIDS_VERSION)
```

Comment:
You could probably simplify this a bit more if you reworked the logic to something like
```
if(_RAPIDS_COMPONENTS)
  # Set common things
  if (_RAPIDS_VERSION)
    # set version and some string to stick in the components filename.
  else()
    ...
  endif()
  configure_file...
```